back/engines/commercial/report_builder/app/models/report_builder/patches/user.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module ReportBuilder::Patches::User
+  def self.included(base)
+    base.class_eval do
+      has_many :reports, class_name: 'ReportBuilder::Report', foreign_key: 'owner_id', dependent: :destroy
+    end
+  end
```

Comment:
So this means that when a user is deleted, all their reports are as well right?

I do wonder if on a live platform this is what we want? It means if an admin leaves a municipality & they remove the user account from their CitizenLab platform, any reports they've created will also be removed right? :thinking:

Reply:
Yes indeed, I was hesitating but since the owner is mandatory, this seemed to be the right choice. @alexander-cit Could you perhaps confirm that it's fine to delete the reports when deleting the owner, or would it be better to make the owner optional and keep the reports when deleting the owner?

Reply:
It's a good question. I guess indeed it's better to leave reports when a user is deleted. It does require more changes and some testing, but it'll be safer.